### PR TITLE
Add wildcard inclusion/exclusion support to container names

### PR DESCRIPTION
**Pull request by m4ce**
This PR solves #2734 .

### CODE REVIEW

**Comment by m4ce** [comment]
If changes are okay, I will go ahead implementing the tests.

**Comment by danielnelson** [comment]
What are the possible states other than running?

**Comment by m4ce** [comment]
@danielnelson, I guess the possible states are: paused, restarting, removing, running, dead, created, exited.

I suppose it only makes sense to track containers in running state? I'd no reason to to extend this to containers in states other than running?

**Comment by m4ce** [comment]
@danielnelson - let me know if the changes look good. Once they do, I will take care of the test units.

File: plugins/inputs/docker/README.md [hunk 1]
```diff
@@ -20,14 +20,22 @@ for the stat structure can be found
   ##   To use TCP, set endpoint = "tcp://[ip]:[port]"
   ##   To use environment variables (ie, docker-machine), set endpoint = "ENV"
   endpoint = "unix:///var/run/docker.sock"
+
   ## Only collect metrics for these containers, collect all if empty
   container_names = []
+
+  ## Containers to include and exclude. Globs accepted.
+  ## Note that the options are ignored when using the container_names option. An empty array for both will include all containers
```
Comment (danielnelson) [inline plugins/inputs/docker/README.md]: We should favor the new options, maybe you can just append `container_names` to the new value?  Then document that the old version is deprecated.  Include the version it is deprecated in and the replacement, something like: "Deprecated (1.4.0), use container_name_include"

**Comment by m4ce** [comment]
@danielnelson - I've made the changes. If they look good, I will proceed with the test cases.

**Comment by danielnelson** [comment]
Looks good

**Comment by m4ce** [comment]
@danielnelson - I've added the tests. It's ready to merge.

**Comment by danielnelson** [comment]
Perfect!

**Pull request merged**